src/ladds/distributedMemParallelization/RankMigration.cpp

```diff
@@ -0,0 +1,198 @@
+/**
+ * @file RankMigration.cpp
```

Comment:
This file needs a rename. Migrating ranks typically refers to moving an MPI rank to a different compute node which is not what we are doing here! Maybe `ParticleCommunication` or something like this?

Reply:
There is already the ParticleCommunicator which makes it very confusing to have ParticleCommunication. Opted for "ParticleMigrationHandler" ?